### PR TITLE
Announcing Enquirer 2.0! - Complete Refactor

### PR DESCRIPTION
**Pull request by jonschlinkert**
# Announcing Enquirer 2.0!

This version of Enquirer is a complete overhaul, not only in terms of how prompts are written and handled but also in terms of our philosophy for creating and maintaining prompts. 

- Enquirer 2.0 will ship with several built-in prompts (some prompt libraries will be deprecated)
- Semantic styling system that makes colors and symbols easier than every to customize
- Only one dependency!
- Easier to customize and use
- New prompt types 
- New features

### CODE REVIEW

File: examples/select/option-symbols.js [hunk 1]
```diff
@@ -18,21 +16,12 @@ const prompt = new Prompt({
...
-      let fallback = status === 'on' ? '🗡️ ' : '  ';
-      if (typeof symbol === 'function') {
-        return symbol(...arguments) || fallback;
-      }
-      return symbol || fallback;
+    pointer(state) {
+      return ['🍬', '🍎', '👄', '🖕', '🗡️ '][state.index];
     },
+    prefix(state) {
+      return ({ pending: '🎃', cancelled: '⚰️ ', answered: '💀' })[state.status];
```
Comment (doowb) [inline examples/select/option-symbols.js]: This is probably one of my favorite examples 🎃

**Comment by linkjay** [comment]
Got me on the edge of my seat with this one!

Can't wait for the merge.

File: examples/select/option-symbols.js (hunk 1 above)
Comment (jonschlinkert) [reply examples/select/option-symbols.js]: Yeah that one is fun :)

**Comment by tunnckoCore** [comment]
:100: :tada:

**Pull request merged**